ui/apps/platform/src/services/AdministrationEventsService.ts

```diff
@@ -86,9 +105,84 @@ export function getAdministrationEvent(id: string): Promise<AdministrationEvent>
         .then((response) => response.data.event);
 }
 
-// TODO ListAdministrationEventsRequest
-export function listAdministrationEvents(): Promise<AdministrationEvent[]> {
+export function listAdministrationEvents(
+    arg: ListAdministrationEventsRequest
+): Promise<AdministrationEvent[]> {
+    const params = qs.stringify(arg, { arrayFormat: 'repeat', allowDots: true });
     return axios
-        .get<ListAdministrationEventsResponse>(eventsUrl)
+        .get<ListAdministrationEventsResponse>(`${eventsUrl}?${params}`)
         .then((response) => response.data.events);
 }
+
+// Given searchFilter from useURLSearch hook, return validated filter argument.
+
+export function getAdministrationEventsFilter(
+    searchFilter: SearchFilter
+): AdministrationEventsFilter {
+    // For consistency with useURLSort hook, especially in case the table columns become sortable,
+    // useURLSearch hook also uses search strings.
+    // See proto/storage/administration_event.proto
+    return {
+        domain: getValue(searchFilter['Event Domain']),
+        level: getLevel(searchFilter['Event Level']),
+        resourceType: getValue(searchFilter['Resource Type']),
+        type: getType(searchFilter['Event Type']),
+    };
+}
+
+type SearchFilterValue = string | string[] | undefined;
+
+function getLevel(arg: SearchFilterValue): AdministrationEventLevel[] | undefined {
+    if (typeof arg === 'string' && isLevel(arg)) {
+        return [arg];
+    }
+
+    if (Array.isArray(arg)) {
+        return arg.filter((item) => isLevel(item)) as AdministrationEventLevel[];
```

Comment:
Is the `as AdministrationEventLevel[]` type assertion needed here? I thought `filter` correctly inferred types.

If so, does it work omitting the extra arrow function and using it like `return arg.filter(isLevel);`?

Reply:
Yes, super. It does here and below for `isType` predicate.